Update README for pool mining on mainnet with self select (#2480)
tari-project/tari: README.md

- For the Tari Merge Mining Proxy, under section **`merge_mining_proxy.ridcully`**
  ```
  [merge_mining_proxy.ridcully]
  monerod_url = "http://18.132.124.81:18081"
  proxy_host_address = "127.0.0.1:7878"
  monerod_use_auth = false
  monerod_username = ""
  monerod_password = ""
  ```

> **Note:** None of the IP address + port combinations listed above should be in use otherwise. 

Reviewer: The previous version of this statement might be taken to include the `18.132...` addresses as well.
```suggestion
**Note:** The ports `7878`, `18142` and `18143` should not be in use by other processes. If they are, choose different ports. You will need to update the ports in the steps below as well. 
```

[here](https://github.com/tari-project/tari/blob/development/applications/tari_merge_mining_proxy/monero_stagenet_setup.md)._

#### XMRig configuration

The XMRig configuration must be prepared for either solo or pool merged mining with Monero. It is advisable to use a 
configuration file for XMRig as this offers more flexibility, otherwise, the configuration parameters can be passed 
in via the command line upon runtime.

**Note:** Monero mainnet and stagenet wallet addresses can only be used with the corresponding network. The `monerod_url` 
configuration setting (see [Tari components](#tari-components)) must also correspond to the chosen network.

##### Solo mining

The [XMRig configuration wizard](https://xmrig.com/wizard) can be used to create a solo mining configuration file 
in JSON format:

- Start -> `+ New configuration`

- Pools -> `+ Add daemon`
  - With `Add new daemon for Solo mining`, complete the required information, then `+ Add daemon`:
     - `Host`, `Port`: This must correspond to the `proxy_host_address` in the Tari configuration file.
     - `Secure connection (TLS)`: Uncheck.
     - `Coin`: Monero.

>      - `Wallet address`: This must be your own stagenet or mainnet wallet address. As an example:

Reviewer: ```suggestion
     - `Wallet address`: This must be your own stagenet or mainnet wallet address, or you can use these donation addresses:
```

    ]
}
```
##### Pool mining with self select

For pool mining, the configuration file obtained from the [XMRig configuration wizard](https://xmrig.com/wizard) must 
be augmented with Tari specific settings. Using the wizard, create the following:

- Start -> `+ New configuration`

- Pools -> `+ Add pool` -> `Custom pool`

  - With `Add new custom pool`, complete the required information, then `+ Add pool`:
    - `Host`, `Port`: This must be for a Monero mainnet mining pool that supports the `self-select`.
    - `Secure connection (TLS)`: Check/Uncheck (based on the pool requirements).
    - `keepalive`: Check.
    - `nicehash`: Uncheck.

>     - `User`: This must be your own mainnet wallet address. As an example:

Reviewer: ```suggestion
    - `User`: This must be your own mainnet wallet address, or you can use this address to donate to Monero:
```

       - Public mainnet address at https://www.getmonero.org/get-started/contributing/
         `888tNkZrPN6JsEgekjMnABU4TBzc2Dt29EPAvkRxbANsAnjyPbb3iQ1YBRk1UXcdRsiKc9dhwMVgN5S9cQUiyoogDavup3H`
    - `Password`: A custom field that could be your wallet name or some other pool settings.
    - `Coin`: Monero.
    - `Algorithm`: rx/0.

- Backends -> Select `CPU` (`OpenCL` or `CUDA` also possible depending on your computer hardware).

- Misc -> With `Donate`, type in your preference.

- Result -> With `Config file`, copy or download, than save as `config.json`.

- Add custom entries for `"self-select": "127.0.0.1:7878"` and `"submit-to-origin": true` in the `"pools"` section.

The resulting configuration file should look like this:

```
{
    "autosave": true,
    "cpu": true,
    "opencl": false,
    "cuda": false,
    "pools": [
        {
            "coin": "monero",
            "algo": "rx/0",
            "url": "cryptonote.social:5555",

>             "user": "888tNkZrPN6JsEgekjMnABU4TBzc2Dt29EPAvkRxbANsAnjyPbb3iQ1YBRk1UXcdRsiKc9dhwMVgN5S9cQUiyoogDavup3H.donatemonero",

Reviewer: ```suggestion
            "user": "<REPLACE WITH YOUR ADDRESS>.<Pool specific user name>",
```

Author: I have used the gist of what you proposed here, i.e. made it clear that the wallet address should be the user's own wallet address, but also kept the example configuration file as shown previously. Please see updated for your further review.

The resulting configuration will look like this:

```
{
    "autosave": true,
    "cpu": true,
    "opencl": false,
    "cuda": false,
    "pools": [
        {
            "coin": "monero",
            "algo": "rx/0",
            "url": "monerop.com:4343",
            "user": "489r43gR8bDMJNBf4Q6sL9CNERvZQrTqjRCSESqgWQEWWq2UGAfj2voaw3zBtD7U8CQ391Nc1PDHUHiN85yhbZnCDasqzyX",
            "pass": "MyWallet",
            "tls": true,
            "keepalive": true,
            "nicehash": false,
            "self-select": "127.0.0.1:7878"
        }
    ]
}
```

#### Merge Mining Proxy configuration

The `monerod_url` field in the `config.toml` should be replaced by the mainnet value:
 
```
# URL to monerod, currently set to a seed node
monerod_url = "http://18.132.124.81:18081" # mainnet
```

#### Runtime

When running all components as previous ([Perform merge mining](#perform-merge-mining)), look out for the following 
outputs in the XMRig console to confirm that it is connected to the pool and submitting its own custom template to the
`monerod_url`:

```
* POOL #1      monerop.com:4343 coin monero self-select 127.0.0.1:7878
```

> Similat snippets should be visible in the proxy log file:  

Reviewer: ```suggestion
Similar snippets should be visible in the proxy log file:  
```
